integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/AppendDataDatasetBuilder.java

```diff
@@ -0,0 +1,51 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.api.AbstractQueryPlanOutputDatasetBuilder;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.spark.sql.catalyst.plans.logical.AppendData;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+
+/**
+ * {@link LogicalPlan} visitor that matches an {@link AppendData} commands and extracts the output
+ * {@link OpenLineage.Dataset} being written.
+ */
+@Slf4j
+public class AppendDataDatasetBuilder extends AbstractQueryPlanOutputDatasetBuilder<AppendData> {
+
+  private final DatasetFactory<OpenLineage.OutputDataset> factory;
+
+  public AppendDataDatasetBuilder(
+      OpenLineageContext context, DatasetFactory<OpenLineage.OutputDataset> factory) {
+    super(context);
+    this.factory = factory;
+  }
+
+  @Override
+  public boolean isDefinedAtLogicalPlan(LogicalPlan logicalPlan) {
+    return logicalPlan instanceof AppendData;
+  }
+
+  @Override
+  public List<OpenLineage.OutputDataset> apply(AppendData x) {
+    // Needs to cast to logical plan despite IntelliJ claiming otherwise.
+    LogicalPlan logicalPlan = (LogicalPlan) ((AppendData) x).table();
+
+    return context.getOutputDatasetBuilders().stream()
+        .filter(b -> b.getClass().getName().equals("DataSourceV2RelationOutputDatasetBuilder"))
+        .flatMap(b -> b.apply(logicalPlan).stream())
+        .collect(Collectors.toList());
+
+    // FIXME: need to do something
+    // NEED TO FIX IT;
+    //    return PlanUtils.applyFirst(
+    //        context.getOutputDatasetBuilders(), (LogicalPlan) ((AppendData) x).table()
+    //    );
+  }
```

Comment:
The `AbstractQueryPlanDatasetBuilder#delegate` method should be what you're looking for here.